models/articles.models.js

```diff
@@ -40,4 +40,18 @@ exports.selectAllCommentsByArticleId=((article_id)=>{
         } 
         return result.rows
     })
+})
+
+exports.postACommentOnArticle=((article_id, username, body)=>{
+    return db.query(`INSERT INTO comments
+        (article_id, author, body)
+        VALUES ($1, $2, $3) RETURNING *
+        ;`,[article_id, username, body])
+
+        .then((result)=>{
+            if(result.rows.length===0){
+                return Promise.reject({status: 404, msg: "Error 404 - Article Not Found"})
```

Comment:
Do we need this custom error here?
From what I can tell, your tests check for a PSQL error (article not found) which should be caught by your app middleware.